# 📊 NovaMart Retail Sales - Exploratory Data Analysis (EDA)

## Project Overview

This notebook performs exploratory data analysis on the NovaMart Retail Sales dataset generated using Python.

The objective is to understand sales performance, customer behavior and business trends before creating SQL reports and Power BI dashboards.

### Objectives

- Understand dataset structure
- Calculate important business KPIs
- Analyze sales performance
- Identify top-performing products and cities
- Study payment trends
- Prepare data for visualization

# Import Libraries

In [3]:
import pandas as pd
import matplotlib.pyplot as plt

# Step 1 - Load Dataset

In [4]:
df = pd.read_csv("novamart_sales_dataset.csv")

df.head()

,order_id,order_date,customer_id,customer_name,city,state,region,category,product_name,quantity,unit_price,discount_percent,sales,cost,profit,payment_method,salesperson
0,ORD000001,2025-07-01,CUST4066,Daksha Puri,Vijayawada,Andhra Pradesh,South,Electronics,iPhone 16,1,85000,15,72250.0,72000,250.0,Credit Card,hrithik anand
1,ORD000002,2025-03-04,CUST1819,Balendra Purohit,Bangalore,Karnataka,South,Accessories,Logitech Mouse,3,900,0,2700.0,1950,750.0,UPI,Rahul Sharma
2,ORD000003,2025-11-30,CUST2844,Pahal Rajagopalan,Mysore,Karnataka,South,Furniture,Bookshelf,4,8500,0,34000.0,24800,9200.0,UPI,Divya dharshini
3,ORD000004,2025-12-13,CUST4751,Vyanjana Divan,Hyderabad,Telangana,South,Electronics,iPad Air,1,58000,0,58000.0,49000,9000.0,Net Banking,Rahul Sharma
4,ORD000005,2025-03-06,CUST2478,Gavin Agate,Thiruvananthapuram,Kerala,South,Clothing,Jacket,2,4200,10,7560.0,6000,1560.0,Net Banking,Sneha Reddy


# Step 2 - Understand the Dataset

In [5]:
df.shape

print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Rows: 10000
Columns: 17


In [6]:
df.columns

Index(['order_id', 'order_date', 'customer_id', 'customer_name', 'city',
       'state', 'region', 'category', 'product_name', 'quantity', 'unit_price',
       'discount_percent', 'sales', 'cost', 'profit', 'payment_method',
       'salesperson'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   order_id          10000 non-null  object 
 1   order_date        10000 non-null  object 
 2   customer_id       10000 non-null  object 
 3   customer_name     10000 non-null  object 
 4   city              10000 non-null  object 
 5   state             10000 non-null  object 
 6   region            10000 non-null  object 
 7   category          10000 non-null  object 
 8   product_name      10000 non-null  object 
 9   quantity          10000 non-null  int64  
 10  unit_price        10000 non-null  int64  
 11  discount_percent  10000 non-null  int64  
 12  sales             10000 non-null  float64
 13  cost              10000 non-null  int64  
 14  profit            10000 non-null  float64
 15  payment_method    10000 non-null  object 
 16  salesperson       10000 non-null  object 

## Observation

- Dataset contains 10,000 transactions.
- No missing values.
- Data types are appropriate for analysis.

In [8]:
df.describe()

,quantity,unit_price,discount_percent,sales,cost,profit
count,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000
mean,2.999900,18020.755000,9.996000,48989.004025,44413.924000,4575.080025
std,1.404171,24785.595985,7.105634,78218.512494,72708.656309,9119.797449
min,1.000000,65.000000,0.000000,52.000000,48.000000,-20000.000000
25%,2.000000,1100.000000,5.000000,2800.000000,2000.000000,355.000000
50%,3.000000,4200.000000,10.000000,10500.000000,9000.000000,1300.000000
75%,4.000000,31000.000000,15.000000,61100.000000,55000.000000,4500.000000
max,5.000000,85000.000000,20.000000,425000.000000,360000.000000,65000.000000


# Step 3 - Business KPIs

Calculation of
  1. Total Sales
  2. Total Profit
  3. Average Order Value

In [9]:
total_sales = df["sales"].sum()
total_profit = df["profit"].sum()
average_order = df["sales"].mean()

print(f"💰 Total Sales : ₹{total_sales:,.2f}")
print(f"📈 Total Profit : ₹{total_profit:,.2f}")
print(f"🛒 Average Order Value : ₹{average_order:,.2f}")

💰 Total Sales : ₹489,890,040.25
📈 Total Profit : ₹45,750,800.25
🛒 Average Order Value : ₹48,989.00


### Observation

- The dataset generated total sales of ₹ 489,890,040
- The business achieved a total profit of ₹ 45,750,800
- The average order value indicates the typical customer spending per transaction.

# Step 4 - Product Category Analysis

Which category earns the most?

In [10]:
df.groupby("category")["sales"].sum().sort_values(ascending=False)

category
Electronics        2.948642e+08
Furniture          8.988878e+07
Home Appliances    8.480536e+07
Clothing           1.130100e+07
Accessories        7.208815e+06
Grocery            1.821935e+06
Name: sales, dtype: float64

### Insight

Electronics generated the highest sales revenue.

This suggests customers spend more on high-value electronic products than other categories.

# Step 5 - Payment Method Analysis

Which payment method is most popular?

In [11]:
df["payment_method"].value_counts()

payment_method
UPI            4534
Credit Card    1974
Debit Card     1463
Cash           1017
Net Banking    1012
Name: count, dtype: int64

### Insight

UPI was the most preferred payment method, indicating customers prefer digital transactions.

# Step 6 - City-wise Sales Analysis

Which city generated the highest sales?

In [12]:
df.groupby("city")["sales"].sum().sort_values(ascending=False)

city
Coimbatore            56398823.75
Mysore                52988224.00
Pune                  52859798.00
Bangalore             49599514.50
Chennai               48314260.75
Vijayawada            47329750.25
Hyderabad             46975215.25
Thiruvananthapuram    45414232.50
Kochi                 45114885.75
Madurai               44895335.50
Name: sales, dtype: float64

### Insight

Coimbatore generated the highest revenue, making it the strongest performing market.

# Step 7 - Salesperson Performance

Who generated the most revenue?

In [13]:
df.groupby("salesperson")["sales"].sum().sort_values(ascending=False)

salesperson
Priyanka           51804975.25
Rohan Patel        51069286.25
Arjun Kumar        50973004.75
Rahul Sharma       50882472.50
Arsheya begum      50186616.50
hrithik anand      49346861.50
Sneha Reddy        48546747.00
Ananya pandya      46324241.00
Vikram Singh       46136836.00
Divya dharshini    44618999.50
Name: sales, dtype: float64

### Insight

The highest-performing salesperson Priyanka generated the maximum sales revenue and can be studied to understand best sales practices.

# Step 8 - Monthly Sales Analysis

In [14]:
df["order_date"] = pd.to_datetime(df["order_date"])

monthly_sales = (
    df.groupby(df["order_date"].dt.month_name())["sales"]
      .sum()
      .reindex([
          "January","February","March","April",
          "May","June","July","August",
          "September","October","November","December"
      ])
)

monthly_sales

order_date
January      42300730.75
February     35211098.75
March        37396503.00
April        40586213.75
May          40741323.00
June         43861077.50
July         38555276.25
August       44192144.00
September    40037320.25
October      41887611.25
November     45146449.25
December     39974292.50
Name: sales, dtype: float64

### Insight

Highest Sales Month : November

Lowest Sales Month : February

# Step 9 - Top Product Analysis

In [15]:
df.groupby("product_name")["sales"].sum().sort_values(ascending=False).head(10)

product_name
iPhone 16             83372250.0
Lenovo ThinkPad       72766200.0
Dell Inspiron 15      55369100.0
iPad Air              50544100.0
Sofa Set              40047750.0
Refrigerator          36459100.0
Samsung Galaxy A56    32812500.0
Washing Machine       27934100.0
Dining Table          25873400.0
Microwave Oven        13568100.0
Name: sales, dtype: float64

### Insight

The products listed above contributed the highest sales revenue and represent the company's strongest-performing items.

# Step 10 - Customers Analysis

In [16]:
df.groupby("customer_name")["sales"].sum().sort_values(ascending=False).head(10)

customer_name
Ekantika Om         909720.0
Abeer Bhat          902426.0
Raagini Raval       874740.0
Samarth Yohannan    870750.0
Ayushman Shanker    849700.0
Fitan Kuruvilla     813250.0
Shravya Bahri       802320.0
Ekbal Mandal        801000.0
Pooja Gala          717400.0
Oscar Luthra        698459.0
Name: sales, dtype: float64

### Insight

These customers generated the highest sales value and could be considered valuable repeat customers.

# Step 11 - Profitability Analysis

In [17]:
df.groupby("category")["profit"].mean().sort_values(ascending=False)

category
Electronics        11103.374413
Furniture           7072.810438
Home Appliances     6036.035294
Clothing            1763.039927
Accessories          968.334363
Grocery              179.701598
Name: profit, dtype: float64

### Insight

This analysis identifies which product category provides the highest average profitability rather than just the highest sales.

# Step 12 - Average Order Value Analysis

In [18]:
df.groupby("payment_method")["sales"].mean().sort_values(ascending=False)

payment_method
Debit Card     50592.973342
Credit Card    50018.980370
UPI            48608.411226
Cash           48548.543756
Net Banking    46808.940217
Name: sales, dtype: float64

### Insight

Customers using Debit card tend to place higher-value orders on average.

## Key Findings

Total Sales :
₹ 489890040

Total Profit :
₹ 45750800

Highest Revenue Category :
Electronics

Top City :
Coimbatore

Most Used Payment Method :
UPI

Top Salesperson :
Priyanka

Highest Profit Category :
Electronics

## Conclusion

The exploratory data analysis provided valuable insights into NovaMart's sales performance, customer purchasing behavior, product profitability, and regional trends.

These findings will be used to build SQL reports and an interactive Power BI dashboard.